### PR TITLE
first version of implementation of AccountService class and related unit tests

### PR DESCRIPTION
**Pull request by Steve3029**

### CODE REVIEW

**Comment by unruledboy** [comment]
Coding standards / styles
	* 1 blank line between fields and constructor
	* 1 blank line between functions
	* operator at the front

		* request == null

                        || string.IsNullOrWhiteSpace(request.Email)                        || string.IsNullOrWhiteSpace(request.Password)
	* prefer conversion to enum than int because easier to auto complete

		* no

			* if (account.Status == (int)AccountStatus.Inactivated)
		* yes

			* if ((AccountStatus)account.Status == AccountStatus.Inactivated) //but, if Status is not a valid AccountStatus, it is just int value
	* naming

		* InvalidAccount: it is an function to verify something, it should be IsInvalidAction
		* AccountStatus

			* Inactivated -> Inactive
			* Activated -> Active
	* magic numbers / strings

		* Issuer = "DevYeah"
		* Expires = DateTime.UtcNow.AddHours(12)

Design / usage
	* leaky abstraction

		* _repository.Add(newAccount), then _repository.SaveChanges()
		* AccountRepository.GetAccountByEmail

			* account.Email == email -> unique index on Email
	* abstraction

		* sending email:

			* _smtpClient, should be in dedicated service
			* should not use named parameters unless necessary
			* values like sender should be from configuration
			* subject / body can be in template (think localization)
			* _smtpClient.SendMailAsync() returns a task, you should await it
		* class Model

			* is it really necessary for only the Id property?
			* It would stop proper inheritance
	* comparison

		* for equality, just use == , rather than like account.Password.Equals(sha256Pws)
	* MailServer -> MailService ?

Encapsulation
	* password hashing logic should be wrapped, rather than repeated and separated GetMd5Hash then GetSha256Hash
	* the logic in one function is too complication, try to separate them into smaller functions, like SignIn/SignUp
	* GetMd5/GetSha256Hash/BuildHexadecimalString etc. potentially can be moved to a utililty class
	* In AccountServiceTest

		* "region resolve config file path" likely to be used in other test classes, maybe we should have a base test class?
		* GenerateToken should not be here, it should only be in production code
		* TestSignInFailed, it should initialize the dummy account first, rather than SignUp then SignIn

Spelling
	* "let he/she" -> "let him/her'
	* OfficalEmailAddress -> OfficialEmailAddress

**Comment by Steve3029** [comment]
The code of business service has been refactored. And wrapping email related code is a bit time consuming, it will be rewritten at next commit.

File: DatabaseInitialScript.sql [hunk 1]
```diff
@@ -175,6 +175,11 @@ IF NOT EXISTS (SELECT NULL FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = 'C
 	  ADD [Level] TINYINT NOT NULL
   GO
 
+IF NOT EXISTS (SELECT NULL FROM sys.indexes WHERE NAME='Idx_Email')
```
Comment (unruledboy) [inline DatabaseInitialScript.sql]: db object naming convention: [Coding Standard](https://github.com/devyeah/LMS/wiki/Coding-Standard)

**Comment by unruledboy** [comment]
@Steve3029 Identityhelper -> Identity**H**elper

**Pull request merged**